gems/sorbet-runtime/lib/types/props/decorator.rb

```diff
@@ -302,6 +294,24 @@ def prop_validate_definition!(name, cls, rules, type)
     T::Utils::Nilable.is_union_with_nilclass(cls) || ((cls == T.untyped || cls == NilClass) && rules.key?(:default) && rules[:default].nil?)
   end
 
+  sig { params(name: Symbol).returns(T::Boolean) }
+  private def is_override?(name)
+    @class.method_defined?(name) && !@class.method_defined?(name, false)
+  end
```

Comment:
I don't think that this is strictly additive, because of how this interacts with `without_accessors`

The old version, where it would check `props.include?(prop)` would also fire if the parent definition was a prop with `without_accessors: true`.

`without_accessors` is very weird, and I'd love to not relax the checks we're doing if we don't need to. It's totally fine to put the `method_defined?` checks first so that we short circuit earlier, but I think we should still be looking in `props` as a fallback.